porch/repository/pkg/git/package.go

```diff
@@ -147,12 +146,12 @@ func (p *gitPackageRevision) GetUpstreamLock() (kptfile.Upstream, kptfile.Upstre
 }
 
 func (p *gitPackageRevision) getPackageRevisionLifecycle() v1alpha1.PackageRevisionLifecycle {
-	switch {
-	case p.ref == nil || p.ref.Name().IsTag():
+	switch ref := p.ref; {
+	case ref == nil:
 		return v1alpha1.PackageRevisionLifecycleFinal
-	case strings.HasPrefix(p.ref.Name().String(), refDraftPrefix):
+	case isDraftBranchNameInLocal(ref.Name()):
```

Comment:
Idea for the future - we could have a "parse ref" function which returns a struct including the kind (draft / proposed etc), the name etc.  I think we could also have the inverse function (struct -> ref name).

Reply:
Yeah ... I've tried several experiments on this one and the current incarnation uses custom string types, but I was quite close creating struct already.
I am already moving all of the mapping logic into one location and I expect we'll turn it into a more generic component. At that point we'll need to up-level the abstraction too, most likely.